exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb

```diff
@@ -107,11 +109,13 @@ def fetch_ssl_verify_mode
             end
           end
 
-          def http_connection(uri, ssl_verify_mode, certificate_file)
+          def http_connection(uri, ssl_verify_mode, certificate_file, client_certificate_file, client_key_file)
             http = Net::HTTP.new(uri.host, uri.port)
             http.use_ssl = uri.scheme == 'https'
             http.verify_mode = ssl_verify_mode
             http.ca_file = certificate_file unless certificate_file.nil?
+            http.cert = OpenSSL::X509::Certificate.new(File.read(client_certificate_file)) unless client_certificate_file.nil?
```

Comment:
Especially when running in containers, and using environment variables, it would be nice to be able to load the PEM encoded cert/key directly from environment variables, instead of having to find a way to write it to a file first.

Reply:
I agree but at least one file will still be needed for many/most users because it's assigned to `Net::HTTP#ca_file` which can only be a path. Should I make the change to support either for the client cert and key despite that?

Reply:
I think that would make it inconsistent with other SDK implementations which all use file paths. 

I think that if we have a request to do this we may want to steer folks in a direction to generate a file via a configMap or equivalent.